contracts/rewards/src/contract/execute.rs

```diff
@@ -0,0 +1,184 @@
+use std::collections::HashMap;
+
+use axelar_wasm_std::nonempty::Uint256;
+use cosmwasm_std::Addr;
+
+use crate::{
+    error::ContractError,
+    msg::RewardsParams,
+    state::{Epoch, Store},
+};
+
+pub struct Contract<S>
+where
+    S: Store,
+{
+    pub store: S,
+}
+
+#[allow(dead_code)]
+impl<S> Contract<S>
+where
+    S: Store,
+{
+    /// Returns the current epoch. The current epoch is computed dynamically based on the current
+    /// block height and the epoch duration. If the epoch duration is updated, we store the epoch
+    /// in which the update occurs as the last checkpoint
+    fn get_current_epoch(&mut self, cur_block_height: u64) -> Result<Epoch, ContractError> {
+        let stored_params = self.store.load_params();
+        let epoch_duration: u64 = stored_params.params.epoch_duration.into();
+        let epoch = stored_params.last_updated;
+        if cur_block_height >= epoch.block_height_started + epoch_duration {
+            let new_epoch_num = ((cur_block_height - epoch.block_height_started) / epoch_duration)
+                + epoch.epoch_num;
+            let new_epoch = Epoch {
+                epoch_num: new_epoch_num,
+                block_height_started: cur_block_height - (cur_block_height % epoch_duration),
```

Comment:
this is not accurate as soon as the epoch_duration param changes

Reply:
you need to do `(cur_block_height - epoch.blockc_height_started) % epoch_duraction`, because we can only look ata time window when the duration has been fixed